distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/mock/csw/ChunkedContent.java

```diff
@@ -0,0 +1,158 @@
+     * continue waiting for another retry attempt from the calling server.
+     */
+    public void resetFailState() {
+        numberOfRetries = 0;
+    }
+
+    /**
+     * Returns the minimum headers required for sending a chunked data response, but does not
+     * include the response function itself.
+     *
+     * @return the minimum headers required for sending a chunked data response
+     */
+    public static Action getHeaders() {
+        return composite(contentType("text/plain"),
+                header("Transfer-Encoding", "chunked"),
+                header("content-type", "text/plain"));
+    }
+
+    /**
+     * Constructor for a response that has a delay and a number of planned failure.
+     *
+     * @param responseMessage  Message to be sent in response.
+     * @param messageDelay     Time to wait between sending each character of the message.
+     * @param numberOfFailures Number of times to fail (simulate network disconnect) after sending
+     *                         the first character. Once this number is reached, the message will
+     *                         send successfully.
+     */
+    public static Action newResponse(String responseMessage, Duration messageDelay,
+            int numberOfFailures) {
+        return custom(new ChunkedContent(responseMessage, messageDelay, numberOfFailures));
+
+    }
+
+    /**
+     * Returns an composite action that holds the headers required for a plain text response as well
+     * as the response function itself. Unless the headers included are not wanted, this is the
+     * preferred way to set the Restito response actions.
+     * <p>
+     * Note that additional headers may also be needed for specific endpoint functionality. This
+     * method returns a composite action so that additional headers can be set alongside this action.
+     *
+     * @param responseMessage  Message to be sent in response.
+     * @param messageDelay     Time to wait between sending each character of the message.
+     * @param numberOfFailures Number of times to fail (simulate network disconnect) after sending
+     *                         the first character. Once this number is reached, the message will
+     *                         send successfully.
+     * @return composite action that holds the headers required for a plain text response as well
+     * as the response function itself.
+     */
+    public static Action newResponseWithHeaders(String responseMessage, Duration messageDelay,
+            int numberOfFailures) {
+        return composite(getHeaders(),
+                newResponse(responseMessage, messageDelay, numberOfFailures));
+    }
+
+    /**
+     * Implementation of the Function interface's apply method. This class can also be used as a
+     * Function<Response, Response> directly in the Restito response if custom actions are needed.
+     *
+     * @param response
+     * @return
+     */
+    @Override
+    public Response apply(Response response) {
```

Comment:
Something to consider. Maybe we should have a `private` inner-class implement the `Function<>` interface instead of having `ChunkedContent` implement it. That way, the `ChunkedContent` class wouldn't need to have a `public` `apply` method and would only contain the `static` factory methods which would make a better API.